gcloud-java-examples/src/main/java/com/google/gcloud/examples/BigQueryExample.java

```diff
@@ -0,0 +1,686 @@
+ * operations that apply to more than one entity (`list`, `create`, `info` and `delete`) the third
+ * parameter specifies the entity. {@code <primitiveType>} indicates that only primitive types are
+ * supported by the {@code create table} and {@code create external-table} operations
+ * ({@code string}, {@code float}, {@code integer}, {@code timestamp}, {@code boolean}).
+ * {@code <sourceUri>}, {@code <sourceUris>} and {@code <destinationUris>} parameters are URIs to
+ * Google Cloud Storage blobs, in the form {@code gs://bucket/path}. See each action's run method
+ * for the specific BigQuery interaction.
+ */
+public class BigQueryExample {
+
+  private static final Map<String, BigQueryAction> CREATE_ACTIONS = new HashMap<>();
+  private static final Map<String, BigQueryAction> INFO_ACTIONS = new HashMap<>();
+  private static final Map<String, BigQueryAction> LIST_ACTIONS = new HashMap<>();
+  private static final Map<String, BigQueryAction> DELETE_ACTIONS = new HashMap<>();
+  private static final Map<String, BigQueryAction> ACTIONS = new HashMap<>();
+
+  private abstract static class BigQueryAction<T> {
+
+    abstract void run(BigQuery bigquery, T request) throws Exception;
+
+    abstract T parse(String... args) throws Exception;
+
+    protected String params() {
+      return "";
+    }
+  }
+
+  private static class ParentAction extends BigQueryAction<Tuple<BigQueryAction, Object>> {
+
+    private final Map<String, BigQueryAction> subActions;
+
+    public ParentAction(Map<String, BigQueryAction> subActions) {
+      this.subActions = ImmutableMap.copyOf(subActions);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    void run(BigQuery bigquery, Tuple<BigQueryAction, Object> subaction) throws Exception {
+      subaction.x().run(bigquery, subaction.y());
+    }
+
+    @Override
+    Tuple<BigQueryAction, Object> parse(String... args) throws Exception {
+      if (args.length >= 1) {
+        BigQueryAction action = subActions.get(args[0]);
+        if (action != null) {
+          Object actionArguments = action.parse(Arrays.copyOfRange(args, 1, args.length));
+          return Tuple.of(action, actionArguments);
+        }
+      }
+      throw new IllegalArgumentException();
+    }
+
+    @Override
+    public String params() {
+      StringBuilder builder = new StringBuilder();
+      for (Map.Entry<String, BigQueryAction> entry : subActions.entrySet()) {
+        builder.append("\n").append(entry.getKey());
+        String param = entry.getValue().params();
+        if (param != null && !param.isEmpty()) {
+          builder.append(' ').append(param);
+        }
+      }
+      return builder.toString();
+    }
+  }
+
+  private abstract static class VoidAction extends BigQueryAction<Void> {
+    @Override
+    Void parse(String... args) throws Exception {
+      if (args.length == 0) {
+        return null;
+      }
+      throw new IllegalArgumentException();
```

Comment:
message suggesting something like "this operation does not accept any arguments"?